src/ophys_etl/modules/segmentation/postprocess_utils/roi_time_correlation.py

```diff
@@ -0,0 +1,204 @@
+import numpy as np
+from ophys_etl.modules.segmentation.postprocess_utils.roi_types import (
+    SegmentationROI)
+
+
+def sub_video_from_roi(roi: SegmentationROI,
+                       video_data: np.ndarray) -> np.ndarray:
+    """
+    Get a sub-video that is flattened in space corresponding
+    to the video data at the ROI
+
+    Parameters
+    ----------
+    roi: SegmentationROI
+
+    video_data: np.ndarray
+        Shape is (ntime, nrows, ncols)
+
+    Returns
+    -------
+    sub_video: np.ndarray
+        Shape is (ntime, npix) where npix is the number
+        of pixels marked True in the ROI
+    """
+
+    xmin = roi.x0
+    ymin = roi.y0
+    xmax = roi.x0+roi.width
+    ymax = roi.y0+roi.height
+
+    sub_video = video_data[:, ymin:ymax, xmin:xmax]
+
+    mask = roi.mask_matrix
+    sub_video = sub_video[:, mask].reshape(video_data.shape[0], -1)
+    return sub_video
+
+
+def correlate_sub_video(sub_video: np.ndarray,
+                        key_pixel: np.ndarray,
+                        filter_fraction: float = 0.2) -> np.ndarray:
+    """
+    Correlated all of the pixels in a sub_video against
+    a provided time series using only the brightest N
+    timesteps
+
+    Parameters
+    ----------
+    sub_video: np.ndarray
+        Shape is (ntime, npixels)
+
+    key_pixel: np.ndarray
+        Shape is (ntime,)
+        This is the time series against which to correlate
+        the pixels in sub_video
+
+    filter_fraction: float
+        Keep the brightest filter_fraction timesteps when doing
+        the correlation (this is reckoned by the flux values in
+        key_pixel)
+
+    Returns
+    -------
+    corr: np.ndarray
+        Shape is (npix,)
+        These are the correlation values of the pixels in
+        sub_video against key_pixel
+    """
+
+    npix = sub_video.shape[1]
+    discard = 1.0-filter_fraction
+    th = np.quantile(key_pixel, discard)
+    mask = (key_pixel >= th)
+    sub_video = sub_video[mask, :]
+    key_pixel = key_pixel[mask]
+    ntime = len(key_pixel)
+    assert key_pixel.shape == (ntime,)
```

Comment:
don't need asserts in code, should be covered by tests

Reply:
removed